studio/components/interfaces/BillingV2/Usage/Infrastructure.tsx

```diff
@@ -0,0 +1,127 @@
+import ShimmeringLoader from 'components/ui/ShimmeringLoader'
+import { useInfraMonitoringQuery } from 'data/analytics/infra-monitoring-query'
+import { useProjectSubscriptionQuery } from 'data/subscriptions/project-subscription-query'
+import dayjs from 'dayjs'
+import BarChart from './BarChart'
+import SectionContent from './SectionContent'
+import SectionHeader from './SectionHeader'
+import { USAGE_CATEGORIES } from './Usage.constants'
+
+export interface InfrastructureProps {
+  projectRef: string
+}
+
+// [Joshen] Need to update the IO budget chart to show burst mbps and duration next time
+
+const Infrastructure = ({ projectRef }: InfrastructureProps) => {
+  const { data: subscription } = useProjectSubscriptionQuery({ projectRef })
+  const { current_period_start, current_period_end } = subscription?.billing ?? {}
+  const startDate = new Date((current_period_start ?? 0) * 1000).toISOString()
+  const endDate = new Date((current_period_end ?? 0) * 1000).toISOString()
+  const categoryMeta = USAGE_CATEGORIES.find((category) => category.key === 'infra')
+
+  const { data: cpuUsageData, isLoading: isLoadingCpuUsageData } = useInfraMonitoringQuery({
```

Comment:
Same here, might make sense for the endpoint to support multiple attributes to avoid issuing so many requests, not sure if this would be as easy as our backend usage endpoint though

Reply:
Not a blocker for this PR, just an improvement we can do in the future

Reply:
resolving convo - response [here](https://github.com/supabase/supabase/pull/14118#discussion_r1185034869)